apps/web/src/middleware.ts

```diff
@@ -9,9 +9,16 @@ export function middleware(request: NextRequest) {
     (pathname.startsWith("/signup") || pathname.startsWith("/signin")) &&
     authToken
   )
-    return NextResponse.rewrite(new URL("/", request.url));
+    return NextResponse.redirect(new URL("/", request.url));
 
-  if (!authToken) return NextResponse.rewrite(new URL("/signin", request.url));
+  if (
+    !authToken &&
+    pathname !== "/signin" &&
+    pathname !== "/signup" &&
+    pathname !== "/forgot-password"
```

Comment:
We can create a config file to store array of these routes instead of adding them in if condition directly everytime

Reply:
we will do it later?